src/OrchardCore.Modules/OrchardCore.Workflows/Http/Scripting/HttpMethodsProvider.cs

```diff
@@ -134,18 +137,109 @@ public HttpMethodsProvider(IHttpContextAccessor httpContextAccessor)
                                  select new JProperty(param.Key, JArray.FromObject(param.Value.ToArray()))).ToArray()))
             };
 
+            // This should be deprecated
             _requestFormAsJsonMethod = new GlobalMethod
             {
                 Name = "requestFormAsJson",
                 Method = serviceProvider => (Func<JObject>)(() =>
                     new JObject((from field in httpContextAccessor.HttpContext.Request.Form
                                  select new JProperty(field.Key, JArray.FromObject(field.Value.ToArray()))).ToArray()))
             };
+
+            _requestDataAsJsonObjectMethod = new GlobalMethod
+            {
+                Name = "requestDataAsJsonObject",
```

Comment:
There is no such thing of a JSON object. It's either json or an object. In this case it's a `JObject` (NewtonSoft) but I don't see why it should be it, and not just a `Dictionary<string, object>`. Or does it matter because it's passed back to C# and it's better than a Dictionary?

Reply:
in my case, it was not passed back to C#. It was used only in scripts and liquid. Dictionary would work.

Reply:
would probably even make more sense to call it `deserializeRequestData`/`deserializeRequestForm`

Reply:
Ok last thing remaining to test is `Dictionary<string, object>` instead of `JObject`

Reply:
While it still works there are some things that don't get serialized properly. This is probably why it got changed to send a JObject directly so that it can be formatted the way we want it exactly. I'm going to try to make it work with `Dictionary<string, object>` but it seems to not work as it should.

Reply:
application/x-www-form-urlencoded : 

![image](https://user-images.githubusercontent.com/3228637/119886675-37fafc00-bf01-11eb-8d94-591fc1c24cf4.png)

application/json : 

![image](https://user-images.githubusercontent.com/3228637/119886799-5e209c00-bf01-11eb-8779-599c64b12583.png)

We can see here that with application/json we can pass fields with single string values <string, string>
But with application/x-www-form-urlencoded we can't pass fields with a single string value because of IDictionary<string, object>.

I think it is fine enough since application/json seems more appropriate for sending "custom json" to the API.